File: first-semester/Programming-Fundamentals/labs/Strings/SubTextoMejorado.java
import java.util.Scanner;
public class Main {
  public static void main(String[] args) {
    Scanner scan = new Scanner(System.in);
    String texto1 = scan.next();
    int posicionInicial = scan.nextInt();
    int posicionFinal = scan.nextInt();
    if (posicionFinal > texto1.length()) {
      System.out.println("Error");
    } else if (posicionInicial > posicionFinal) {
      System.out.println("Error2");
    } else {
      System.out.println(texto1.substring(posicionInicial, (posicionFinal + 1)));
    }
  }
}

File: 2-semester/Data-structure-and-algorithms-I/labs/seguimiento-0/cadenas/subtextomejorado.py
palabra = input()
posicion_inicial = int(input())
posicion_final = int(input())

if posicion_final > len(palabra):
    print("Error")
elif posicion_inicial > posicion_final:
    print("Error2")
else:
    print(palabra[posicion_inicial:posicion_final+1])